src/components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,190 @@
+import { removeProperty } from "../util/RemoveProperty.js";
+import { TestUtils } from "../util/TestUtils.js";
+
+const Dialog = ({
+  children,
+  className,
+  id = Rivet.shortuid(),
+  align,
+  closeOnOutsideClickOrEscape,
+  darkenPage,
+  disablePageInteraction,
+  isOpen,
+  onDismiss,
+  showCloseButton = true,
+  title,
+  ...attrs
+}) => {
+  const ref = React.useRef(null);
+
+  useEffect(() => {
+    handleEventRegistration();
+    return () => {
+      clickOutsideOrEscapeEventHandler.deregister();
+    };
+  }, [isOpen]);
+
+  const { dialogProps, titleProps } = useDialog(
+    { id, ["aria-labelledby"]: `${id}-title`, role: "dialog", ...attrs },
+    ref
+  );
+
+  const { modalProps, underlayProps } = useModalOverlay(
+    {
+      isDismissable: false, // using clickOutsideOrEscapeEventHandler
+      isKeyboardDismissDisabled: false, // using clickOutsideOrEscapeEventHandler
+      attrs,
+    },
+    {
+      isOpen: isOpen,
+      disablePageInteraction: disablePageInteraction,
+    },
+    ref
+  );
+
+  const handleClickOutsideOrEscape = (event) => {
+    if (event && shouldCloseDialog(event)) {
+      handleDismiss();
+      event.stopPropagation && event.stopPropagation();
+    }
+  };
+
+  const clickOutsideOrEscapeEventHandler = handler(handleClickOutsideOrEscape);
+
+  const shouldCloseDialog = (event) => {
+    // If closing is disabled, the user right clicks, presses an unhandled key, presses tab or clicks in the dialog then do not close
+    return !(
+      isRightMouseClick(event) ||
+      isUnhandledKeyPress(event) ||
+      isTabKeyPress(event) ||
+      targets(ref.current, event)
+    );
+  };
+
+  const handleEventRegistration = () => {
+    if (isOpen && closeOnOutsideClickOrEscape) {
+      clickOutsideOrEscapeEventHandler.register();
+    } else {
+      clickOutsideOrEscapeEventHandler.deregister();
+    }
+  };
+
+  const dataRvt = {
+    ...{ ...(align && { [`data-rvt-dialog-${align}`]: "" }) },
+    ...{ ...(darkenPage && { ["data-rvt-dialog-darken-page"]: "" }) },
+  };
+
+  const handleDismiss = () => {
+    onDismiss && onDismiss();
+  };
+
+  const dialogContent = (overlayProps) => (
+    <div
+      className={classNames("rvt-dialog", className)}
+      aria-hidden={!isOpen}
+      hidden={!isOpen}
+      ref={ref}
+      {...dataRvt}
+      {...overlayProps}
+    >
+      {title && (
+        <header
+          className="rvt-dialog__header"
+          data-testid={TestUtils.Dialog.dialogHeaderTestId}
+        >
+          <h1
+            className="rvt-dialog__title"
+            id={`${id}-title`}
+            {...removeProperty(titleProps, "id")}
+          >
+            {title}
+          </h1>
+        </header>
+      )}
+      {children}
+      {onDismiss && showCloseButton && (
+        <DialogCloseButton onDismiss={handleDismiss} />
+      )}
+    </div>
+  );
+
+  if (isOpen) {
+    if (disablePageInteraction) {
+      return (
```

Comment:
```suggestion
      // If page interaction is disabled, we need to render the dialog content inside a fixed div which takes up the entire screen, otherwise just render the dialog content
      return (
```